docs/tool-choice.md

```diff
@@ -0,0 +1,348 @@
+# Tool Choice
+
+Tool choice controls which tools the model can use during a request. This is useful for controlling agent behavior, optimizing for specific use cases, or working around provider limitations.
+
+## Overview
+
+PydanticAI distinguishes between two types of tools:
+
+- **Function tools**: Tools you register with the agent using `@agent.tool` or `tools=[...]`
+- **Output tools**: Internal tools used by the framework for structured output (e.g., `final_result`)
+
+The `tool_choice` setting in [`ModelSettings`][pydantic_ai.settings.ModelSettings] controls which **function tools** are available to the model. Output tools are handled separately to ensure agents can always complete their work.
+
+## Options
```

Comment:
It's a bit hard now to find what the options are and when you'd use each, as you need to read multiple section headers and bodies. I think this doc can be simplified a lot, but introducing the feature in one paragraph, then having a list of all options with a description of what they do, and finally the limitations to keep in mind + one (or a few) examples